src/modules/sensors/sensors.cpp

```diff
@@ -507,7 +509,7 @@ Sensors::adc_poll(struct sensor_combined_s &raw)
 							selected_source = b;
 
 							/* Move the selected_source to instance 0 */
-							if (_battery_pub_intance0ndx != b) {
+							if (BOARD_NUMBER_BRICKS > 1 && _battery_pub_intance0ndx != b) {
```

Comment:
@davids5 yes sorry for the lack of explanation. 
This is a fix for a compiler warning in the gcc7 CI build newly added by @dagar in #7644
```
../src/modules/sensors/sensors.cpp:512:54: error: array subscript is outside array bounds [-Werror=array-bounds]
         tmp_h = _battery_pub[_battery_pub_intance0ndx];
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
compilation terminated due to -Wfatal-errors.
cc1plus: all warnings being treated as errors
[170/535] Building CXX object src/modu.../modules__sensors.dir/parameters.cpp.o
ninja: build stopped: subcommand failed.
Makefile:158: recipe for target 'posix_sitl_default' failed
```
The full build log is here: https://travis-ci.org/PX4/Firmware/jobs/255098609

I think the compiler is trying to be too smart. If this block is executed, then _battery_pub_instance0ndx will be out the f bounds. 
Because _battery_pub_instance0ndx is initialized to 0 and never modified elsewhere the block cannot be executed, but apparently the compiler cannot see that.

I tried to move the initialization of the variable to the cpp file, but that did not help the compiler. I left it there because this is done like this elsewhere, but I can revert if you prefer.